File: src/main/java/cz/stasimek/fakturaceeasypeasy/entity/InvoiceItem.java
package cz.stasimek.fakturaceeasypeasy.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import java.math.BigDecimal;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.springframework.validation.annotation.Validated;

@Entity
@SQLDelete(sql = "UPDATE APP.invoice_item SET deleted = true WHERE id = ?")
@Getter
@Setter
@Validated
public class InvoiceItem extends BaseEntity {

	@JsonBackReference
	@NotNull
	@ManyToOne(optional = false)
	private Invoice invoice;

	@Column
	private BigDecimal quantity;

	@Size(max = 20)
	@Column(length = 20)
	private String unit;

	@NotNull
	@Size(max = 255)
	@Column(length = 255, nullable = false)
	private String description;

	@NotNull
	@Column(nullable = false)
	private BigDecimal unitPrice;

	// Sazba DPH (%)
	@NotNull
	@Column(nullable = false)
	private BigDecimal vatRate;

}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/entity/InvoiceItem.kt
package cz.stasimek.fakturaceeasypeasy.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import org.hibernate.annotations.SQLDelete
import org.springframework.validation.annotation.Validated
import java.math.BigDecimal
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@SQLDelete(sql = "UPDATE APP.invoice_item SET deleted = true WHERE id = ?")
@Validated
open class InvoiceItem : BaseEntity() {

    @JsonBackReference
    @ManyToOne(optional = false)
    @NotNull
    open var invoice: Invoice? = null

    @Column
    open var quantity: BigDecimal? = null

    @Column(length = 20)
    @Size(max = 20)
    open var unit: String? = null

    @Column(length = 255, nullable = false)
    @NotNull
    @Size(max = 255)
    open var description: String? = null

    @Column(nullable = false)
    @NotNull
    open var unitPrice: BigDecimal? = null

    // Sazba DPH (%)
    @Column(nullable = false)
    @NotNull
    open var vatRate: BigDecimal? = null
}